tests/test_cluster.py

```diff
@@ -0,0 +1,659 @@
+        a QNodeCluster with the correct QNodes"""
+        dev_list = [qml.device("default.qubit", wires=1), qml.device("default.qubit", wires=1)]
+
+        obs_list = [qml.PauliX(0), qml.PauliY(0)]
+        template = lambda x, wires: qml.RX(x, wires=0)
+
+        qc = qml.map(template, obs_list, dev_list)
+
+        assert len(qc) == 2
+
+        # evaluate cluster so that queue is populated
+        qc(1)
+
+        assert len(qc[0].ops) == 2
+        assert qc[0].ops[0].name == "RX"
+        assert qc[0].ops[1].name == "PauliX"
+
+        assert len(qc[1].ops) == 2
+        assert qc[1].ops[0].name == "RX"
+        assert qc[1].ops[1].name == "PauliY"
+
+        # test that device is not broadcast
+        assert qc[0].device is not qc[1].device
+        assert qc[0].device is dev_list[0]
+        assert qc[1].device is dev_list[1]
+
+    def test_mapping_over_measurements(self):
+        """Test that mapping over a list of measurement types produces
+        a QNodeCluster with the correct QNodes"""
+        dev = qml.device("default.qubit", wires=1)
+
+        obs_list = [qml.PauliX(0), qml.PauliY(0)]
+        template = lambda x, wires: qml.RX(x, wires=0)
+
+        qc = qml.map(template, obs_list, dev, measure=["expval", "var"])
+
+        assert len(qc) == 2
+
+        # evaluate cluster so that queue is populated
+        qc(1)
+
+        assert len(qc[0].ops) == 2
+        assert qc[0].ops[0].name == "RX"
+        assert qc[0].ops[1].name == "PauliX"
+        assert qc[0].ops[1].return_type == qml.operation.Expectation
+
+        assert len(qc[1].ops) == 2
+        assert qc[1].ops[0].name == "RX"
+        assert qc[1].ops[1].name == "PauliY"
+        assert qc[1].ops[1].return_type == qml.operation.Variance
+
+    def test_invalid_obserable(self):
+        """Test that an invalid observable raises an exception"""
+        dev = qml.device("default.qubit", wires=1)
+
+        obs_list = [qml.PauliX(0), qml.S(wires=0)]
+        template = lambda x, wires: qml.RX(x, wires=0)
+
+        with pytest.raises(ValueError, match="Some or all observables are not valid"):
+            qml.map(template, obs_list, dev, measure=["expval", "var"])
+
+
+class TestApply:
+    """Tests for the apply function"""
+
+    @pytest.mark.parametrize("interface", ["autograd", "numpy", "torch", "tf"])
+    def test_apply_summation(self, qnodes, interface, tf_support, torch_support, tol):
+        """Test that summation can be applied using all interfaces"""
+        if interface == "torch" and not torch_support:
+            pytest.skip("Skipped, no torch support")
+
+        if interface == "tf" and not tf_support:
+            pytest.skip("Skipped, no tf support")
+
+        qnode1, qnode2 = qnodes
+        qc = qml.QNodeCluster([qnode1, qnode2])
+
+        if interface == "tf":
+            sfn = tf.reduce_sum
+        elif interface == "torch":
+            sfn = torch.sum
+        else:
+            sfn = np.sum
```

Comment:
This could also be passed as an argument

Reply:
Do you mean passed like the following?
```python
@pytest.mark.parametrize("interface, sfn",
    [("autograd", np.sum), ("numpy", np.sum), ("torch", torch.sum), ("tf", tf.reduce_sum)])
```

Unfortunately this won't work, since `torch.sum` and `tf.reduce_sum` won't be defined if PyTorch or TF aren't installed. So they can only be called _after_ the `pytest.skip` statements.

Reply:
You're right, slipped my attention! Thanks!